tools/keyboard/scancodes_LCID_files_generator.cpp

```diff
@@ -0,0 +1,393 @@
+                       Mods::CAPSLOCK_MOD | Mods::SHIFT_MOD},
+                Layout{layout.capslock_altGr,
+                       Mods::CAPSLOCK_MOD | Mods::ALTGR_MOD},
+                Layout{layout.capslock_shiftAltGr,
+                       Mods::CAPSLOCK_MOD | Mods::SHIFT_MOD | Mods::ALTGR_MOD}
+            })
+        {
+            /* don't generate with numpad scancodes because
+               there are already ones */
+            auto is_numpad = [](std::uint8_t scancode)
+            {
+                return scancode >= 0x47 && scancode <= 0x53;
+            };
+            std::uint8_t scancode = 0;
+
+            for (std::uint16_t unicode : wrapped_layout.klayout)
+            {
+                if (unicode_occurences.find(unicode)
+                    == unicode_occurences.cend())
+                {
+                    if (!is_numpad(scancode))
+                    {
+                        ofs << "    { ";
+                        code_hex16_stream_fmt(unicode);
+                        ofs << ", { ";
+                        code_hex8_stream_fmt(scancode);
+                        ofs << ", ";
+                        code_hex8_stream_fmt
+                            (std::uint8_t(wrapped_layout.mods));
+                        ofs << " } },\n";
+                        
+                        unicode_occurences.emplace(unicode);
+                    }
+                }
+                ++scancode;
+            }
+        }
+    
+        ofs << "  };\n\n";
+
+
+    
+        /* Write extended unicode to 
+           extended scancode + unicode conversion table */
+        ofs << "  // key : extended unicode, value : extended scancode + unicode [8 + 16 bits]\n"
+            << "  inline const std::unordered_map<std::uint16_t, std::pair<std::uint8_t, std::uint16_t>> EXTENDED_SCANCODES_TABLE\n"
+            << "  {\n";
+
+        std::unordered_set<std::uint16_t> extended_unicode_occurences;
+        
+        for (std::uint8_t i = 0; i < layout.nbDeadkeys; ++i)
+        {
+            for (std::uint8_t j = 0; j < layout.deadkeys[i].nbSecondKeys; ++j)
+            {
+                std::uint16_t extended_unicode =
+                    layout.deadkeys[i].secondKeys[j].modifiedKey;
+
+                if (extended_unicode_occurences.find(extended_unicode)
+                    == extended_unicode_occurences.cend())
+                {
+                    ofs << "    { ";
+                    code_hex16_stream_fmt(extended_unicode);
+                    ofs << ", { ";
+                
+                    std::uint8_t extended_scancode =
+                        layout.deadkeys[i].extendedKeyCode;
+                    std::uint16_t unicode =
+                        layout.deadkeys[i].secondKeys[j].secondKey;
+                
+                    code_hex8_stream_fmt(extended_scancode);
+                    ofs << ", ";
+                    code_hex16_stream_fmt(unicode);
+                    ofs << " } },\n";
```

Comment:
With `code_hex8_stream_fmt` and `code_hex16_stream_fmt` as "I/O manipulator":

```cpp
ofs << "    { " << hex16_fmt(extended_unicode) << ", { "
    << hex8_fmt(extended_scancode) << ", " << hex16_fmt(unicode)
    << " } },\n";
```

See `Hex8` and `Hex16` in `jsclient/tool/reversed_keylayout_generator.cpp`.